test/functional.jl

```diff
@@ -219,3 +219,12 @@ end
 let (:)(a,b) = (i for i in Base.:(:)(1,10) if i%2==0)
     @test Int8[ i for i = 1:2 ] == [2,4,6,8,10]
 end
+
+@testset "indexing" begin
```

Comment:
While you're at it, it could make sense to test `find*` functions too, as they rely on indexing.

Reply:
I'll leave this out of this PR as it appears there have been other efforts to support this which have stalled: https://github.com/JuliaLang/julia/pull/34678